### PR TITLE
docs(caching): updating the cache configuration with redis

### PR DESCRIPTION
**Pull request by Tony133**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/nestjs/docs.nestjs.com/issues/2689 
Closing this PR: https://github.com/nestjs/docs.nestjs.com/pull/2691

## What is the new behavior?
Updating the cache configuration with Redis

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: content/techniques/caching.md [hunk 1]
```diff
@@ -223,32 +223,39 @@ class HttpCacheInterceptor extends CacheInterceptor {
 
 #### Different stores
 
-This service takes advantage of [cache-manager](https://github.com/node-cache-manager/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis store](https://github.com/dabroek/node-cache-manager-redis-store). A full list of supported stores is available [here](https://github.com/node-cache-manager/node-cache-manager#store-engines). To set up the Redis store, simply pass the package together with corresponding options to the `register()` method.
+This service takes advantage of [cache-manager](https://github.com/node-cache-manager/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis store](https://github.com/node-cache-manager/node-cache-manager-redis-yet) official package for node-cache-manager. A full list of supported stores is available [here](https://github.com/node-cache-manager/node-cache-manager#store-engines). To set up the Redis store, requires `registerAsync()` , since the store factory is initializing the connection to redis and only then returning the instance:
```
Comment (kamilmysliwiec) [inline content/techniques/caching.md:226]:
```suggestion
This service takes advantage of [cache-manager](https://github.com/node-cache-manager/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis store](https://github.com/node-cache-manager/node-cache-manager-redis-yet) official package for node-cache-manager. A full list of supported stores is available [here](https://github.com/node-cache-manager/node-cache-manager#store-engines). To set up the Redis store, use the `registerAsync()` method to initialize the store, as follows:
```

File: content/techniques/caching.md [hunk 2]
```diff
@@ -223,32 +223,39 @@ class HttpCacheInterceptor extends CacheInterceptor {
...
+        const store = await redisStore({
+          socket: {
+            host: 'localhost',
+            port: 6379,
+          },
+        });
+
+        return {
+          store: store as unknown as CacheStore,
+          ttl: 60 * 60 * 24 * 7,  // 1 week
```
[the remark this answers is not in the document]
Comment (Tony133) [reply content/techniques/caching.md:248]:
```suggestion
          ttl:  5 * 1000 // milliseconds
``` 
With the cache manager v5 the TTL and in milleseconds we do 5 * 1000 with written here: https://github.com/node-cache-manager/node-cache-manager  ?

**Review by rubiin — APPROVED** [review]

File: content/techniques/caching.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tony133) [reply content/techniques/caching.md:248]:
```suggestion
          ttl: 3 * 60000,  // 3 minutes (milliseconds)
```

[the remark this answers is not in the document]
Comment (Tony133) [reply content/techniques/caching.md:248]: I set it to 3 minutes

**Review by rubiin — APPROVED** [review]

File: content/techniques/caching.md [hunk 3]
```diff
@@ -223,32 +223,39 @@ class HttpCacheInterceptor extends CacheInterceptor {
 
 #### Different stores
 
-This service takes advantage of [cache-manager](https://github.com/node-cache-manager/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis store](https://github.com/dabroek/node-cache-manager-redis-store). A full list of supported stores is available [here](https://github.com/node-cache-manager/node-cache-manager#store-engines). To set up the Redis store, simply pass the package together with corresponding options to the `register()` method.
+This service takes advantage of [cache-manager](https://github.com/node-cache-manager/node-cache-manager) under the hood. The `cache-manager` package supports a wide-range of useful stores, for example, [Redis store](https://github.com/node-cache-manager/node-cache-manager-redis-yet) official package for node-cache-manager. A full list of supported stores is available [here](https://github.com/node-cache-manager/node-cache-manager#store-engines). To set up the Redis store, use the `registerAsync()` method to initialize the store, as follows:
 
 ```typescript
 import type { RedisClientOptions } from 'redis';
```
Comment (armujahid) [inline content/techniques/caching.md:229]: `RedisClientOptions` isn't being used

Comment (Tony133) [reply content/techniques/caching.md]: Updated, thanks for reporting and reviewing!

**Review by jmcdo29 — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
Merging in it as a temporary improvement - cache-manager@6 was released and it yet again switched to a different package

**Pull request merged**